packages/core/http/core-http-server-internal/src/static_assets/static_assets.test.ts

```diff
@@ -0,0 +1,117 @@
+    });
+  });
+
+  describe('#getPluginAssetHref()', () => {
+    it('returns the expected value when CDN config is not set', () => {
+      cdnConfig = CdnConfig.from();
+      staticAssets = new StaticAssets(basePath, cdnConfig, '');
+      expect(staticAssets.getPluginAssetHref('foo', 'path/to/img.gif')).toEqual(
+        '/base-path/plugins/foo/assets/path/to/img.gif'
+      );
+    });
+
+    it('returns the expected value when CDN config is set', () => {
+      cdnConfig = CdnConfig.from({ url: 'https://cdn.example.com/test' });
+      staticAssets = new StaticAssets(basePath, cdnConfig, '');
+      expect(staticAssets.getPluginAssetHref('bar', 'path/to/img.gif')).toEqual(
+        'https://cdn.example.com/test/plugins/bar/assets/path/to/img.gif'
+      );
+    });
+
+    it('removes leading slash from the assetPath', () => {
+      cdnConfig = CdnConfig.from();
+      staticAssets = new StaticAssets(basePath, cdnConfig, '');
+      expect(staticAssets.getPluginAssetHref('dolly', '/path/for/something.svg')).toEqual(
+        '/base-path/plugins/dolly/assets/path/for/something.svg'
+      );
+    });
+  });
+
+  describe('with a SHA digest provided', () => {
+    describe('cdn', () => {
+      it.each([
+        ['https://cdn.example.com', 'https://cdn.example.com/beef'],
+        ['https://cdn.example.com:1234', 'https://cdn.example.com:1234/beef'],
+        ['https://cdn.example.com:1234/roast', 'https://cdn.example.com:1234/roast/beef'],
+        [
+          'https://cdn.example.com:1234/roast?thing=1',
+          'https://cdn.example.com:1234/roast/beef?thing=1',
+        ],
+      ])('suffixes the digest to the CDNs path value (%s)', (url, expectedHref) => {
+        cdnConfig = CdnConfig.from({ url });
+        staticAssets = new StaticAssets(basePath, cdnConfig, 'beef');
+        expect(staticAssets.getHrefBase()).toEqual(expectedHref);
+      });
+    });
+
+    describe('base path', () => {
+      it.each([
+        ['', '/beef'],
+        ['/', '/beef'],
+        ['/roast', '/roast/beef'],
+      ])('suffixes the digest to the server base path (%s)', (url, expectedPath) => {
+        cdnConfig = CdnConfig.from();
+        basePath = new BasePath(url);
+        staticAssets = new StaticAssets(basePath, cdnConfig, 'beef');
+        expect(staticAssets.getHrefBase()).toEqual(expectedPath);
+      });
+    });
+  });
+
+  describe('#getPluginServerPath()', () => {
+    it('provides the path plugin assets can use for server routes', () => {
+      cdnConfig = CdnConfig.from();
+      staticAssets = new StaticAssets(basePath, cdnConfig, '1234');
+      expect(staticAssets.getPluginServerPath('myPlugin', '/fun/times')).toEqual(
+        '/1234/plugins/myPlugin/assets/fun/times'
+      );
+    });
+  });
+  describe('#appendPathToPublicUrl()', () => {
+    it('with a CDN it appends as expected', () => {
+      cdnConfig = CdnConfig.from({ url: 'http://cdn.example.com/cool?123=true' });
```

Comment:
NIT: I'm not sure we care (or even should support) that awkward thingy of defining a query param in the cdn url and then preserving it when appending paths, do we? 

(but probably irrelevant because surely no one will ever use it, right, right?)

Reply:
++ this is done mainly to show _we could_, but I added a test to the http config that effectively disallows it. Will update this test.